src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs

```diff
@@ -193,5 +186,29 @@ private ITypeSymbol GetDelegateType(TypeInferenceInfo typeInferenceInfo, Compila
 
             return typeSymbol.GetDelegateType(compilation);
         }
+
+        private bool IsRangeExpression(SemanticModel semanticModel, SyntaxToken token, CancellationToken cancellationToken)
+        {
+            if (!token.IsKind(SyntaxKind.DotToken) || !(token.Parent is MemberAccessExpressionSyntax memberAccessExpression))
+            {
+                return false;
+            }
+
+            // On walking up, we should meet BracketedArgumentList before Statement. 
+            // Otherwise, it is not a range expression.
```

Comment:
this is a very weak check.  For example, ranges are allowed in basically any expression context. And, unlike lambdas, they don't need a specific type at that location for inference.  i.e. it is *expected* that someone may write:  ```Foo(a..b)```.  Having completion mess with this would be concerning.  

A better check is "is this a location we *expect* a range.  That will at least deal with the `Foo(a..b)` case.  However, it won't deal with the legal case of writing `var x = a..b;`.  It may be that that case isn't important enough, or that we need to hear about feedback about it before doing something.

Reply:
Yes, if we can provide a more delicate approach, we should wider it something like this:
If there is nothing after dot and the type of the symbol before the dot is numeric, we should:
1. ignore completion if a second dot is typed immediately after.
2. proceed with a regular completion, if any other character was typed after.

Please correct me if you see more conditions to consider.

---
In reply to: [284523908](https://github.com/dotnet/roslyn/pull/35749#discussion_r284523908) [](ancestors = 284523908)